server/CQRItems.cpp

```diff
@@ -190,4 +191,96 @@ void CQRItems::info() {
 	log(irId, "info", ttm);
 }
 
+void CQRItems::isRegion() {
+    auto irId = genIntReqId("isRegion");
+
+    sserialize::TimeMeasurer ttm;
+    ttm.begin();
+
+    // set up environment
+    const auto gh = d().completer->store().geoHierarchy();
+    const auto cqs = request().get("q");
+    const auto cqr = d().completer->cqrComplete(cqs, d().treedCQR, d().treedCQRThreads);
+    auto const idxStore = d().completer->indexStore();
+    const auto subGraph = d().completer->ghsg();
+    const auto store = d().completer->store();
+
+    // find all regions in the query
+    std::unordered_set<uint32_t> regions;
+    std::unordered_set<uint32_t> cqrCells;
+    for (sserialize::CellQueryResult::const_iterator it(cqr.begin()), end(cqr.end()); it != end; ++it) {
+        const auto &cellParents = subGraph.cellParents(it.cellId());
+        cqrCells.insert(it.cellId());
+        for (const uint32_t cellParent : cellParents) {
```

Comment:
Mal sehen ob das ankommt. Das ist mit der vscode Github-PR extension

Reply:
hier sollte eigentlich auch gehen:
```C++
regions.insert(cellParents.begin(), cellParents.end());
```
Der Check `regions.contains(cellParent)` muss ja auch ein look-up in der Hashtable machen. `regions.insert` macht dann nochmal genau das selbe. Da spart man sich also nicht wirklich was.